### PR TITLE
Stylising confirmation boxes

### PR DESCRIPTION
**Pull request by matiasgarciaisaia**
CC: @bcardiff 
See #503

### CODE REVIEW

File: app/assets/stylesheets/_buttons.scss [hunk 1]
```diff
@@ -153,6 +153,21 @@
   }
 }
 
+.btn-delete {
```
Comment (bcardiff) [inline app/assets/stylesheets/_buttons.scss]: .btn-danger seems a better/less action specific name IMO (ref: http://getbootstrap.com/css/#buttons-options)

File: app/views/sites/_form.html.haml [hunk 2]
```diff
@@ -47,7 +47,7 @@
...
+          = link_to "Delete", @site, method: :delete, data: { confirm: "You're about to permanently delete this site. This action CANNOT be undone. Are you sure you want to proceed?" }, class: 'btn-secondary pull-right', title: 'Delete Site'
         - else
           - # TODO: link to page with Devices tab always active - it's the current default, but may change
-          = link_to "Delete", site_path(@site), data: { confirm: "The site '#{@site.name}' has #{@site.devices.count} associated #{"device".pluralize(@site.devices.count)}. You should delete or reassign them to another site before deleting this one.\n\nWould you like to go to the devices list?" }, class: 'btn-secondary pull-right', title: 'Delete Site'
+          = content_tag 'a', "Delete", data: { confirm: "In order to delete this site you must first reassign the <a href=\"#{site_path(@site)}\">#{@site.devices.count} #{"device".pluralize(@site.devices.count)}</a> assigned to it.", confirm_title: 'Action required', confirm_hide_cancel: 'true', confirm_button_message: 'Understood' }, class: 'btn-secondary pull-right', title: 'Delete Site'
```
Comment (bcardiff) [inline app/views/sites/_form.html.haml]: this `<a>` does not have `href` attribute. Without that, browser may apply other style, without appropriate cursor. 
Add `href='#'` or `href='javascript:'`

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -136,3 +136,52 @@ $(document).ready(function(){
   });
 
 });
+
+$.rails.allowAction = function(link){
```
Comment (bcardiff) [inline app/assets/javascripts/application.js]: The whole changes in application.js are well isolated. I would vote to move them all to other file like `styled_confirmation.js`

File: app/views/subscribers/_form.html.haml [hunk 4]
```diff
@@ -53,4 +53,4 @@
       = f.submit nil, class: "btn-primary"
       = link_to 'Cancel', subscribers_path, class: 'btn-link'
       - if @editing
-        = link_to "Delete", subscriber, method: :delete, data: { confirm: "Are you sure you want to delete the location '#{subscriber.name}'" }, class: 'btn-secondary pull-right'
+        = link_to "Delete", subscriber, method: :delete, data: { confirm: "You're about to permanently delete this subscriber. This action CANNOT be undone. Are you sure you want to proceed?" }, class: 'btn-secondary pull-right'
```
Comment (bcardiff) [inline app/views/subscribers/_form.html.haml]: I bet all delete links could be generalized to a helper having only to specify the entity to delete

**Comment by bcardiff** [comment]
Review completed cc: @matiasgarciaisaia

**Comment by matiasgarciaisaia** [comment]
@bcardiff: Fixed :)

**Pull request merged**